cpmpy/expressions/globalconstraints.py

```diff
@@ -418,10 +418,10 @@ class Cumulative(GlobalConstraint):
         Supports both varying demand across tasks or equal demand for all jobs
     """
     def __init__(self, start, duration, end, demand, capacity):
-        super(Cumulative, self).__init__("cumulative",[start,
-                                                       duration,
-                                                       end,
-                                                       demand,
+        super(Cumulative, self).__init__("cumulative",[flatlist(start),
+                                                       flatlist(duration),
+                                                       flatlist(end),
+                                                       flatlist(demand),
```

Comment:
You should check first if the demand parameter is a list or not. Our implementation of the Cumulative constraint supports using a constant demand, but calling flatlist on an integer (or variable) will result in an error.

Reply:
Well spotted! I added a check before `flatlist` is called and a corresponding test case.